src/transformations/molodensky.cpp

```diff
@@ -230,6 +245,13 @@ static PJ_XYZ forward_3d(PJ_LPZ lpz, PJ *P) {
     } else {
         lpz = calc_standard_params(lpz, P);
     }
+    if( lpz.phi == HUGE_VAL ) {
+        proj_errno_set(P, PJD_ERR_TOLERANCE_CONDITION);
+        point.xyz.x = HUGE_VAL;
+        point.xyz.y = HUGE_VAL;
+        point.xyz.z = HUGE_VAL;
+        return point.xyz;
```

Comment:
You could simplify this a bit by just going `return proj_coord_error()`. The same below.

For reference: https://github.com/OSGeo/proj.4/blob/ea2d6723856036060b067d7b43fbec23003da64c/src/internal.cpp#L65-L69

Reply:
Done